include/dd/Package.hpp

```diff
@@ -577,7 +591,11 @@ namespace dd {
                 matrixMatrixMultiplication.clear();
                 toffoliTable.clear();
                 clearIdentityTable();
-                noiseOperationTable.clear();
+                stochasticNoiseOperationCache.clear();
+
+                densityAdd.clear();
+                densityMul.clear();
+                densityNoise.clear();
```

Comment:
these should also be separate, shouldn't they?
there is no need to collect the matrix tables if only density matrices have been collected.